### PR TITLE
Allow the polarity of the step pulse to be set.

### PR DESCRIPTION
**Pull request by benlaurie**
Note: my editor removes trailing whitespace automatically, hence all the changes that delete it.

This fixes a problem with TB6560 stepper drivers, which miss steps if the step pulse is positive.

### CODE REVIEW

**Review by giseburt — CHANGES_REQUESTED** [review]
We don't need/want to do this per segment, which is what happens here. We would only need to set this once and wouldn't expect it to change after startup except in rare cases (initially setting up a machine).

File: g2core/stepper.h [hunk 1]
```diff
@@ -434,14 +435,17 @@ struct Stepper {
     uint32_t _motor_disable_timeout_ms;     // the number of ms that the timeout is reset to
     stPowerState _power_state;              // state machine for managing motor power
     stPowerMode _power_mode;                // See stPowerMode for values
+	bool _invert_step;                  // Invert the step output (where applicable)
```
Comment (giseburt) [inline g2core/stepper.h]: This should move to StepDirStepper. In this Stepper superclass we try not to assume much about the pins themselves.

I see I made the same mistake in my own PR for enable inversion.

**Comment by giseburt** [comment]
Please see https://github.com/synthetos/g2/commit/cb7bf178e76ea1ce855dd940bfbc37eb8df2b2d4 for example of how to do this. And thank you for doing this!!

**Review by giseburt — APPROVED** [review]
Thank you, Ben! I'll manually merge these changes in and handle the issues causing the TravisCI builds to fail.

**Comment by giseburt** [comment]
This PR is set against `master`, but since this needs tested we will pull this in against `edge`. Between this and the TravisCI failures, this will be merged in manually.
